be/src/exec/write_combined_txn_log.cpp

```diff
@@ -24,4 +27,50 @@ Status write_combined_txn_log(const CombinedTxnLogPB& logs) {
     return tablet_mgr->put_combined_txn_log(logs);
 }
 
+void mark_failure(const Status& status, std::atomic<bool>* has_error, Status* final_status) {
+    if (!has_error->load()) {
+        if (!has_error->exchange(true)) {
+            if (final_status->ok()) {
+                *final_status = status;
+            }
+        }
+    }
+}
+
+std::function<void()> create_txn_log_task(const CombinedTxnLogPB* logs, lake::TabletManager* tablet_mgr,
+                                          std::atomic<bool>* has_error, Status* final_status) {
+    return [logs, tablet_mgr, has_error, final_status]() {
+        try {
+            Status status = tablet_mgr->put_combined_txn_log(*logs);
+            if (!status.ok()) {
+                throw std::runtime_error("Txn log write failed");
+            }
+        } catch (const std::exception& e) {
+            mark_failure(Status::IOError(e.what()), has_error, final_status);
+        } catch (...) {
+            mark_failure(Status::Unknown("Unknown exception in write combined txn log task"), has_error, final_status);
+        }
+    };
+}
+
+Status write_combined_txn_log_parallel(const std::map<int64_t, CombinedTxnLogPB>& txn_log_map) {
+    CountDownLatch latch(0);
+    std::atomic<bool> has_error(false);
+    Status final_status;
+    for (const auto& [partition_id, logs] : txn_log_map) {
+        auto task_logic =
+                create_txn_log_task(&logs, ExecEnv::GetInstance()->lake_tablet_manager(), &has_error, &final_status);
+        auto task = std::make_shared<AutoCleanRunnable>(std::move(task_logic), [&latch]() { latch.count_down(); });
+        latch.add_count(1);
```

Comment:
instead of adding this `add_count` interface, the tasks can be created ahead and saved in a vector, and then try to submit them one by one.

No matter the submit is successful or not, the countdown will be corrected counted along with the task destruction, so just break the loop in submit if encounter a failure and then at the end, wait for count down to zero.

This might be easier to understand rather than increasing the countdown number.